sky/data/storage.py

```diff
@@ -986,6 +998,65 @@ def sync_all_stores(self):
         for _, store in self.stores.items():
             self._sync_store(store)
 
+    def _maybe_compress_sync_store(self, store: AbstractStore):
+        """Same as sync_store, but compresses before uploading"""
+        # Only supports workdir, self.source should be type str
+        if self.source is None or not isinstance(self.source, str):
+            self._sync_store(store)
+            return
+
+        def num_files(source, excluded_list):
+
+            file_count = 0
+            spec = pathspec.PathSpec.from_lines(
+                pathspec.patterns.GitWildMatchPattern, excluded_list)
+            for root, _, files in os.walk(source):
+                for file in files:
+                    if file.startswith('skypilot-filemounts-'
+                                      ) and file.endswith('.tar.gz'):
+                        logger.warning(f'Consider renaming the file: {file}')
```

Comment:
We should directly raise an error here since IIUC this will essentially remove the user file by override them?